core/src/main/scala/org/http4s/headers/Origin.scala

```diff
@@ -76,8 +75,8 @@ object Origin extends HeaderKey.Internal[Origin] with HeaderKey.Singleton {
       .string
       .map(Uri.Scheme.unsafeFromString)
     val stringHost = until1(char(':').orElse(`end`)).map(RegName.apply)
-    val bracketedIpv6 = char('[') *> Rfc3986.ipv6 <* char(']')
-    val host = List(bracketedIpv6, Rfc3986.ipv4, stringHost).reduceLeft(_ orElse1 _)
+    val bracketedIpv6 = char('[') *> Uri.Ipv6Address.parser <* char(']')
+    val host = List(bracketedIpv6, Uri.Ipv4Address.parser, stringHost).reduceLeft(_ orElse1 _)
```

Comment:
I think we should unnest these types from the `Uri`, but that can wait until we merge this to main.